pkg/network/protocols/postgres/telemetry.go

```diff
@@ -22,12 +22,76 @@ const (
 	numberOfBucketsSmallerThanMaxBufferSize = 3
 )
 
+type counterEnum int
+
+const (
+	fulfillTableAndOperation counterEnum = iota + 1
+	fulfillOperationNotFound
+	fulfillTableNameNotFound
+	fulfillTableAndOpNotFound
+)
+
+// resultAwareCounter stores counter when goal was achieved and counter when target not found.
+type resultAwareCounter struct {
+	// countTableAndOperation counts the number of successfully retrieved table name and operation.
+	countTableAndOperation *libtelemetry.Counter
+	// countOperationNotFound counts the number of unsuccessful fetches of the operation.
+	countOperationNotFound *libtelemetry.Counter
+	// countTableNameNotFound counts the number of unsuccessful fetches of the table name.
+	countTableNameNotFound *libtelemetry.Counter
+	// countTableAndOpNotFound counts the number of failed attempts to fetch both the table name and the operation.
+	countTableAndOpNotFound *libtelemetry.Counter
+}
+
+// newResultAwareCounter creates and returns a new instance
+func newResultAwareCounter(metricGroup *libtelemetry.MetricGroup, metricName string, tags ...string) *resultAwareCounter {
+	return &resultAwareCounter{
+		countTableAndOperation:  metricGroup.NewCounter(metricName, append(tags, "fulfilled:table_and_op")...),
+		countOperationNotFound:  metricGroup.NewCounter(metricName, append(tags, "fulfilled:no_operation")...),
+		countTableNameNotFound:  metricGroup.NewCounter(metricName, append(tags, "fulfilled:no_table_name")...),
+		countTableAndOpNotFound: metricGroup.NewCounter(metricName, append(tags, "fulfilled:no_table_no_op")...),
+	}
+}
+
+// add adds delta to the counter of completed results or counter incomplete results.
+func (c *resultAwareCounter) add(delta int64, fulfilled counterEnum) {
+	switch fulfilled {
+	case fulfillTableAndOperation:
+		c.countTableAndOperation.Add(delta)
+	case fulfillOperationNotFound:
+		c.countOperationNotFound.Add(delta)
+	case fulfillTableNameNotFound:
+		c.countTableNameNotFound.Add(delta)
+	case fulfillTableAndOpNotFound:
+		c.countTableAndOpNotFound.Add(delta)
+	default:
+		log.Errorf("Add bucket counter, undefined enum.")
+	}
+}
+
+// get returns the counter value based on the result.
+func (c *resultAwareCounter) get(fulfilled counterEnum) int64 {
+	switch fulfilled {
+	case fulfillTableAndOperation:
+		return c.countTableAndOperation.Get()
+	case fulfillOperationNotFound:
+		return c.countOperationNotFound.Get()
+	case fulfillTableNameNotFound:
+		return c.countTableNameNotFound.Get()
+	case fulfillTableAndOpNotFound:
+		return c.countTableAndOpNotFound.Get()
+	default:
+		log.Errorf("Get bucket counter, undefined enum.")
+		return -1
```

Comment:
why -1?
use the default value `0`
generally, metrics are non negative objects, returning signed int64 does not make sense

Reply:
I've changed it